### ISSUE TITLE
Support aws-iam-authenticator

### ISSUE THREAD

**Issue by ianb-mp** [k0sproject/k0s#4662]
### Describe the solution you would like

It would be good to support [aws-iam-authenticator](https://github.com/kubernetes-sigs/aws-iam-authenticator) to provide another option for api authorization.

AFAIK it is not currently possible to deploy this addon due to k0s's control plane isolation. For example, aws-iam-authenticator requires a [nodeSelector for the nodes running kube-apiserver](https://github.com/kubernetes-sigs/aws-iam-authenticator/blob/774efb85b060370538c2d47576fb3ba3e58b2c38/deploy/example.yaml#L212-L214).

**Comment by twz123** [comment]
If this requires to run on the controllers as a DaemonSet, the only caveat I see is that the k0s controllers need to be run with `--enable-worker`. Did you try that out?

**Comment by juanluisvaladas** [comment]
Hi, In addition that, I think you'll need to label the node manually:

```
$ kubectl label node/controller-0.k0s.lab  node-role.kubernetes.io/master=""
node/controller-0.k0s.lab labeled
$ k0s kc get node --show-labels controller-0.k0s.lab
NAME                   STATUS   ROLES                  AGE     VERSION       LABELS
controller-0.k0s.lab   Ready    control-plane,master   3d23h   v1.30.1+k0s   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/arch=amd64,kubernetes.io/hostname=controller-0.k0s.lab,kubernetes.io/os=linux,node-role.kubernetes.io/control-plane=true,node-role.kubernetes.io/master=,node.k0sproject.io/role=control-plane
```

[This annotation is deprecated](https://kubernetes.io/docs/reference/labels-annotations-taints/#node-role-kubernetes-io-master-taint) so I don't know if it makes sense to label the node automatically anymore.

**Comment by ianb-mp** [comment]
> If this requires to run on the controllers as a DaemonSet, the only caveat I see is that the k0s controllers need to be run with --enable-worker. Did you try that out?

No, I haven't tried that. I saw the [FAQ entry](https://docs.k0sproject.io/v1.30.0+k0s.0/FAQ/#why-doesnt-kubectl-get-nodes-list-the-k0s-controllers)  which recommends against this for prod so I'd discounted it. I'm using OIDC for now, but may come back to IAM authenticator later.

**Comment by twz123** [comment]
I'd also check out if it's actually a requirement to run on controller nodes. I had a quick glance on the repo, and it's not directly obvious to me as to why this shouldn't be working if you'd rewrite the controller-pinned DaemonSet into a regular Deployment.
